sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceManager.scala

```diff
@@ -75,3 +89,107 @@ class DataSourceManager {
     dataSourceBuilders.containsKey(normalize(name))
   }
 }
+
+/**
+ * Data Source V1 default source wrapper for Python Data Source.
+ */
+abstract class PythonDefaultSource
+    extends RelationProvider
+    with SchemaRelationProvider
```

Comment:
Data source write support is in progress: https://github.com/apache/spark/pull/43791

Can we use the codegen approach also to support data source write?

Reply:
We can just implements `CreatableRelationProvider`, and invoke writing down logic via `createRelation`.

Reply:
I think this won't conflict with your change (either if it uses DSv2 or DSv1)